docs/source/en/model_doc/gpt2.md

```diff
@@ -60,6 +60,94 @@ different sizes: small, medium, large, xl and a distilled version of the small c
 
 This model was contributed by [thomwolf](https://huggingface.co/thomwolf). The original code can be found [here](https://openai.com/blog/better-language-models/).
 
+## Training
+
+GPT-2 is a decoder-only Transformer. It is trained using teacher forcing. This means that during training, we feed the ground truth
+previous tokens to the model in order to predict the next one. During training, one needs to provide the input sequences to the model as
+`input_ids`, as well as `labels`. The latter are identical to the `input_ids`. In teacher-forcing style, the model will internally shift the
+logits before calculating the loss. This ensures that the model is trained to predict the next token.
```

Comment:
```suggestion
GPT-2 is a decoder-only Transformer. It is trained using teacher-forcing which means during training, the previous tokens (the ground truth) are fed to the model to predict the next one. During training, you need to provide the input sequences to the model as
`input_ids` as well as `labels`. The latter is identical to `input_ids` because in teacher-forcing, the model internally shifts the
logits before calculating the loss. This ensures that the model is trained to predict the next token.
```